src/depthcloud_encoder.cpp

```diff
@@ -127,12 +133,20 @@ void DepthCloudEncoder::connectCb()
   }
 }
 
-void DepthCloudEncoder::dynReconfCb(depthcloud_encoder::paramsConfig& config, uint32_t level)
+void DepthCloudEncoder::cameraInfoCb(const sensor_msgs::CameraInfoConstPtr& cam_info_msg)
 {
-  boost::lock_guard<boost::mutex> lock(config_params_mutex_);
```

Comment:
Why is this not needed?

Reply:
Because if we go the distance and protect all config params in the cloud callback and the dyn reconf callback, then the `f_` also needs a protection, to be pedantic. So the same mutex will need to be locked from 3 different callbacks. Since encoding a single frame is not a critical operation (the next message will be the correct one) I didn't think it's worth the overhead.

If you think it's necessary I can put them back.

Reply:
I didn't take a detailed look at how threads and callbacks interact, but I saw an `AsyncSpinner` with two threads and it seems that `cloudToDepth` can race with `cameraInfoCb` and `dynReconfCb` no? 

> So the same mutex will need to be locked from 3 different callbacks.

Exactly, add one mutex, and have each of these three methods take it (at least in the critical section part).

> Since encoding a single frame is not a critical operation (the next message will be the correct one) I didn't think it's worth the overhead.

> If you think it's necessary I can put them back.

Not sure I understood this, but if it can race it should be protected.